include/jemalloc/internal/tcache_inlines.h

```diff
@@ -195,11 +195,12 @@ tcache_dalloc_small(tsd_t *tsd, tcache_t *tcache, void *ptr, szind_t binind,
 	}
 
 	if (unlikely(!cache_bin_dalloc_easy(bin, ptr))) {
-		if (unlikely(tcache_bin_disabled(binind, bin,
-		    tcache->tcache_slow))) {
+		tcache_slow_t *tcache_slow = tcache->tcache_slow;
+		if (unlikely(tcache_bin_disabled(binind, bin, tcache_slow))) {
 			arena_dalloc_small(tsd_tsdn(tsd), ptr);
 			return;
 		}
+		tcache_slow->lg_fill_div_offset[binind] = 0;
```

Comment:
Move this into `tcache_bin_flush_small` which is a slow path? Putting it here as inlined may cause unwanted side effects on the fast path.